### PR TITLE
This brings back --filter-workers for the 2.x series. This will be go…

### PR DESCRIPTION
**Pull request by andrewvc**
…ne in the next major version

### CODE REVIEW

File: logstash-core/lib/logstash/agent.rb [hunk 1]
```diff
@@ -134,6 +138,12 @@ def execute
     end
     configure
 
+
+    if filter_workers
+      @logger.warn("--filter-workers is deprecated! Please use --pipeline-workers. This setting will be removed in the next major version!")
+      self.pipeline_workers=filter_workers
```
Comment (jordansissel) [inline logstash-core/lib/logstash/agent.rb]: spacing around `=`

File: logstash-core/lib/logstash/agent.rb [hunk 2]
```diff
@@ -36,6 +36,10 @@ class LogStash::Agent < Clamp::Command
          :attribute_name => :pipeline_batch_delay,
          :default => LogStash::Pipeline::DEFAULT_SETTINGS[:pipeline_batch_delay]
 
+  option ["--filter-workers"], "COUNT",
```
Comment (suyograo) [inline logstash-core/lib/logstash/agent.rb]: Its `--filterworkers` not `--filter-workers`

Comment (suyograo) [inline logstash-core/lib/logstash/agent.rb]:
```
suyog@machine:/tmp/logstash-1.5.5$ bin/logstash --help
Usage:
    /bin/logstash agent [OPTIONS]

Options:
    -f, --config CONFIG_PATH      Load the logstash config from a specific file
                                  or directory.  If a directory is given, all
                                  files in that directory will be concatenated
                                  in lexicographical order and then parsed as a
                                  single config file. You can also specify
                                  wildcards (globs) and any matched files will
                                  be loaded in the order described above.
    -e CONFIG_STRING              Use the given string as the configuration
                                  data. Same syntax as the config file. If no
                                  input is specified, then 'stdin
                                  { type => stdin }' is the default input.
                                  If no output is specified, then  'stdout
                                  { codec => rubydebug } }' is default output.
                                   (default: "")
    -w, --filterworkers COUNT     Sets the number of filter workers to run.
```

**Comment by jordansissel** [comment]
Can you update the commit message to say `Fixes #4669` on a line by itself at the end?

File: logstash-core/lib/logstash/agent.rb [hunk 3]
```diff
@@ -134,6 +138,12 @@ def execute
     end
     configure
 
+
+    if filter_workers
+      @logger.warn("--filter-workers is deprecated! Please use --pipeline-workers. This setting will be removed in the next major version!")
```
Comment (suyograo) [inline logstash-core/lib/logstash/agent.rb]: Add 
>
Please use --pipeline-workers or -w. This setting..

File: logstash-core/lib/logstash/agent.rb (hunk 2 above)
Comment (jordansissel) [inline logstash-core/lib/logstash/agent.rb]: Good catch.

File: logstash-core/locales/en.yml [hunk 4]
```diff
@@ -157,6 +157,8 @@ en:
           Check configuration for valid syntax and then exit.
         pipeline-workers: |+
           Sets the number of pipeline workers to run.
+        filter-workers: |+
```
Comment (suyograo) [inline logstash-core/locales/en.yml]: `filterworkers`

File: logstash-core/lib/logstash/agent.rb (hunk 3 above)
Comment (andrewvc) [inline logstash-core/lib/logstash/agent.rb]: can do.

**Comment by andrewvc** [comment]
@suyograo @jordansissel I believe I have addressed all concerns here with my last force push.

**Comment by suyograo** [comment]
LGTM

**Pull request closed without merging**